Objects/longobject.c

```diff
@@ -5307,12 +5308,8 @@ int_bit_length_impl(PyObject *self)
 static int
 popcount_digit(digit d)
 {
-    /* 32bit SWAR popcount. */
-    uint32_t u = d;
-    u -= (u >> 1) & 0x55555555U;
-    u = (u & 0x33333333U) + ((u >> 2) & 0x33333333U);
-    u = (u + (u >> 4)) & 0x0f0f0f0fU;
-    return (uint32_t)(u * 0x01010101U) >> 24;
+    Py_BUILD_ASSERT(sizeof(digit) <= sizeof(uint32_t));
```

Comment:
This could be written more directly as `PyLong_SHIFT <= 32`, which better captures the actual condition we want to test. (The type used for `digit` could conceivably be larger than 32 bits, but we'll only ever be using `PyLong_SHIFT` bits of it.)

Reply:
I updated the assertion with a comment.